### PR TITLE
feat!: migrate from sequence number to timestamp

### PR DESCRIPTION
**Pull request by afennert**
## Why migrating from sequence numbers to timestamps?
With the current migration sequence number there is a huge flaw for projects with multiple devs who can write migrations.

### How is it working right now?
The current implementation checks for gaps in the sequence and for duplicated migration numbers if one the cases happen it will throw an error and will not run the migrations. At the same time it generates new migrations with the nex sequence number based on the migrations that already exists in the migrations folder.

### What is the issue with the current implementation?
The issue could already happen with only two devs, for example we have an existing project with 0001..0003 existing migrtations. The first dev is working on a new content type  `type  a` and its `subtype` he creates two migrations the first one for the `subtype` and the second for the `type a`, he will get those migration files:
- 0004-create-sub-type-for-type-a.ts
- 0005-create-type-a.ts
In the mean time the second dev is also working on a migration to create `type b` and will get this migration file:
- 0004-create-type-b.ts
If either the first or the second dev is merging the changes to the main branch the other one will get an error since the migration number is not unique anymore and the dev who didn't merge first will have to update the sequence number manually. If there are more devs than two it will only get worse.

### What is the proposed solution to the problem?
To prevent this kind of issue, timestamps should be used instead of sequence numbers. With timestamps an order for importing migrations will still be inplace and it is unlikely that two devs get the same timestamp since they are in milliseconds. For the rare case of duplicate timestamps the check will still be there and the dev has to change the migration manually or generate a new one like before but as already mentioned this case should be unlikely and rare.

### Will there be problems for existing migrations with sequence numbers after changing to timestamps?
The migration from sequence numbers to timestamp shouldn't be a problem and should work with existing migrations that still use sequence numbers since the tool is checking the numbers/timestamps and is sorting them ascending to create an order for importing the migrations.

#### What has changed?
- migrate from sequence number to timestamp
- adjusting and adding new tests after migrating from sequence number to timestamp
- rename function createMigrationEntries to createMigrationEntry since it was misleading

⚠️ I marked this change as a breaking change even if it will not break anything for the tool itself with the old sequence numbers, however it could be that it breaks existing tests in projects if sequence numbers in the old format are expected.

### CODE REVIEW

File: lib/cli/commands/deploy.ts [hunk 1]
```diff
@@ -45,7 +45,6 @@ export const handler = async (args: DeployArgs) => {
   })
 }
 
-// TODO this verification should be executed when deploying as well
```
Comment (afennert) [inline lib/cli/commands/deploy.ts:48]: This function is already called in the deploy handler in `lib/cli/commands/deploy` I guess somone forgot to remove it

File: lib/contentful/migrationEntries.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { ContentfulPartialOptions } from "../types"
 import { getClient } from "./client"
 import { MigrationEntry } from "./types"
 
-export async function createMigrationEntries(
+export async function createMigrationEntry(
```
Comment (afennert) [inline lib/contentful/migrationEntries.ts:9]: renamed since this function is returning a single entry and therefor the name is misleading

**Comment by marquesVF** [comment]
I wonder why the CI is not being executed by this PR 🤔 
Could it be because the PR is coming from a fork?

Not sure if it's related to some recent changes (probably not), maybe some updated dependency. But when I try to run the tests locally I have the following:

Screenshot 2022-08-12 at 11 30 38

**Comment by marquesVF** [comment]
We need to downgrade chalk version to 4.1.2 because:
Screenshot 2022-08-12 at 11 46 34

**Comment by afennert** [comment]
> I wonder why the CI is not being executed by this PR 🤔 Could it be because the PR is coming from a fork?
> 
> It's not related to your changes (maybe some updated dependency) but when I try to run the tests locally I have the following:
> 
> So I'm unable to test it. Not even possible to test the master branch.


I also had issues while I tried to test my local changes with yarn link. The issue is that this tool seems to be not supporting esm modules. Since version 5 of  `chalk` they migrated it to be an ESM only module. I have downgraded chalk for my tests, maybe it makes sense to get to an ealier version of chalk and prohibit any auto dependency updater to update a major version for chalk as long as this project cannot handle ESM-only dependencies.

Would you like me to downgrade the version in this PR?

**Comment by afennert** [comment]
> We need to downgrade chalk version to 4.1.2 because: Screenshot 2022-08-12 at 11 46 34

It seems like you were some seconds faster but the question is still the same:

> Would you like me to downgrade the version in this PR?

**Comment by marquesVF** [comment]
> > We need to downgrade chalk version to 4.1.2 because: Screenshot 2022-08-12 at 11 46 34
> 
> It seems like you were some seconds faster 😄, but the question is still the same:
> 
> > Would you like me to downgrade the version in this PR?

Yes, that would be good :) 

By the way, I tested the changes locally with https://github.com/foobaragency/cf-migration-example
And it worked 👍 

Good job 🚀

File: .github/dependabot.yml [hunk 3]
```diff
@@ -8,3 +8,5 @@ updates:
       time: "08:00"
       timezone: Europe/Berlin
     open-pull-requests-limit: 10
+    ignore:
+      - dependency-name: "chalk"
```
Comment (marquesVF) [inline .github/dependabot.yml:12]: 👍

**Review by marquesVF — APPROVED** [review]

**Pull request merged**